src/optics/focusAtom.ts

```diff
@@ -0,0 +1,74 @@
+import { atom } from 'jotai'
+import * as O from 'optics-ts'
+import type { PrimitiveAtom, SetStateAction, WritableAtom } from '../core/types'
+
+export type FocusAtom = {
```

Comment:
```suggestion
type FocusAtom = {
```

I would like to not export this as this can be changed something else in the future.
Please use `typeof focusAtom` instead.

Oh, it's just internal. So, it might be OK, if you prefer this over `typeof focusAtom`.

Reply:
Yeah `typeof focusAtom` didn't work since it didn't apply the generic type... hmm, but I might just remove `useFocus`

Reply:
removed useFocus and used function overloading instead